src/utils/components/CloudinitModal/utils/useCloudInit.ts

```diff
@@ -63,16 +59,25 @@ export const useCloudInit = (vm: V1VirtualMachine): UseCloudInitValues => {
   }, [cloudInit?.userData]);
 
   const cloudInitVolume: V1Volume = React.useMemo(() => {
-    const cloudInitNoCloud: V1CloudInitNoCloudSource = deleteObjBlankValues({
+    const cloudInitNoBlanks = deleteObjBlankValues({
       userData: convertUserDataObjectToYAML(userData, shouldAddHeader),
       networkData: convertNetworkDataObjectToYAML(networkData),
     });
 
-    return {
-      name: 'cloudinitdisk',
-      cloudInitNoCloud,
-    };
-  }, [userData, shouldAddHeader, networkData]);
+    if (cloudInitVol.cloudInitConfigDrive) {
```

Comment:
cloudInitVol can be undefined? if so, please add cloudInitVol?.cloudInitConfigDrive (same goes to line 74)

Reply:
yes! Solved